pycm/pycm_multilabel_cm.py

```diff
@@ -0,0 +1,215 @@
+        if index not in self.samplewise_cms:
+            cm = ConfusionMatrix(
+                self.actual_vector_multihot[index],
+                self.predict_vector_multihot[index],
+                sample_weight=self.weights)
+            self.samplewise_cms[index] = cm
+            return cm
+        return self.samplewise_cms[index]
+
+    def __str__(self):
+        """
+        Multilabel confusion matrix object string representation method.
+
+        :return: representation as str
+        """
+        return self.__repr__()
+
+    def __repr__(self):
+        """
+        Multilabel confusion matrix object representation method.
+
+        :return: representation as str
+        """
+        return "pycm.MultiLabelCM(classes: " + str(self.classes) + ")"
+
+    def __len__(self):
+        """
+        Multilabel confusion matrix object length method.
+
+        :return: length as int
+        """
+        return len(self.classes)
+
+
+def __mlcm_vector_handler__(
+        mlcm,
+        actual_vector,
+        predict_vector,
+        sample_weight,
+        classes):
+    """
+    Handle multilabel object conditions for vectors.
+
+    :param mlcm: multilabel confusion matrix
+    :type mlcm: pycm.MultiLabelCM object
+    :param actual_vector: actual vector
+    :type actual_vector: python list or numpy array of any stringable objects
+    :param predict_vector: vector of predictions
+    :type predict_vector: python list or numpy array of any stringable objects
+    :param sample_weight: sample weights list
+    :type sample_weight: list
+    :param classes: ordered labels of classes
+    :type classes: list
+    :return: None
+    """
+    if not isinstance(actual_vector, (list, numpy.ndarray)) or not \
+            isinstance(predict_vector, (list, numpy.ndarray)):
+        raise pycmVectorError(VECTOR_TYPE_ERROR)
+    if len(actual_vector) != len(predict_vector):
+        raise pycmVectorError(VECTOR_SIZE_ERROR)
+    if len(actual_vector) == 0 or len(predict_vector) == 0:
+        raise pycmVectorError(VECTOR_EMPTY_ERROR)
+    if classes is not None and len(set(classes)) != len(classes):
+        raise pycmVectorError(VECTOR_UNIQUE_CLASS_ERROR)
+    if isinstance(sample_weight, (list, numpy.ndarray)):
+        mlcm.weights = sample_weight
+
+
+def __mlcm_assign_classes__(
+        mlcm,
+        classes):
+    """
+    Assign multilabel object class.
+
+    :param mlcm: multilabel confusion matrix
+    :type mlcm: pycm.MultiLabelCM object
+    :param classes: ordered labels of classes
+    :type classes: list
+    :return: None
+    """
+    mlcm.classes = classes
+    if classes is None:
+        try:
+            mlcm.classes = sorted(
+                list(
+                    set.union(
+                        *mlcm.actual_vector,
+                        *mlcm.predict_vector)))
+        except TypeError:
+            raise pycmMultiLabelError(NOT_ALL_SET_VECTOR_ERROR)
```

Comment:
I think we should change this error message, or add another condition for input with different types!
Take a look at this example:

```python
from pycm import *
mlcm = MultiLabelCM([{1,"cat"}],[{1,"cat"}])
```

Reply:
I will fix it.

Reply:
I changed the error message to:
```
Class extraction from input failed. Input vectors should be a list of sets with unified types.
```